crates/core/tedge_write/src/bin.rs

```diff
@@ -0,0 +1,160 @@
+// TODO: force `destination_path` to be the first argument in clap
```

Comment:
What's the motivation for that?

- Is this to avoid security issues with bad sudo configuration?
- Or on the contrary, to give the `tedge-config-plugin` the freedom to use the appropriate user and mod for the files.

e.g:

```
`tedge    ALL = (ALL) NOPASSWD: /usr/bin/tedge-write --user mosquitto /etc/*`
```

Reply:
The main reason is to not grant root permissions when writing outside allowed directories. The set of rules one can express in `sudoers` is very limited, in order to successfully match on the path we write to (e.g. `/etc/*`) it has to be the first argument, as we match on the prefix.

Reply:
Ok the issue is not really that options can be given to `tedge-write` before the destination path,
but that the __tedge-agent__ must always use `tedge-write` giving the destination path first so the `sudoers` file can use wild card for the allowed targets.

In other word, I don't see this TODO as urgent.